pkg/metrics/metrics.go

```diff
@@ -267,5 +267,7 @@ func UpdateSubsSyncCounterStorage(sub *olmv1alpha1.Subscription) {
 		counterValues.installedCSV = sub.Status.InstalledCSV
 		counterValues.pkg = sub.Spec.Package
 		counterValues.channel = sub.Spec.Channel
+
+		subscriptionSyncCounters[sub.GetName()] = counterValues
```

Comment:
I don't think it needs to block this bugfix, but there should really be some kind of synchronization around access to `subscriptionSyncCounters`. Would you be interested in making a separate PR with that change?

Reply:
@benluddy sure! I'm curious to see what you mean by synchronization, once we discuss that I'd be happy to open up a PR if needed.